libethcore/Farm.h

```diff
@@ -263,26 +273,29 @@ class Farm: public FarmFace
 				unsigned int tempC = 0, fanpcnt = 0, powerW = 0;
 				if (hwInfo.deviceIndex >= 0) {
 					if (hwInfo.deviceType == HwMonitorInfoType::NVIDIA && nvmlh) {
-						wrap_nvml_get_tempC(nvmlh, hwInfo.deviceIndex, &tempC);
-						wrap_nvml_get_fanpcnt(nvmlh, hwInfo.deviceIndex, &fanpcnt);
+						int typeidx = hwInfo.deviceIndex % m_cuda_count;
```

Comment:
Do the mapping here ocl->nvml, but we need a additional field in hwInfo, if the deviceIndex source is CUDA or OpenCL. In case of cuda i would also use the  cuda->nvml  mapping to be on the safe side.